gnssanalysis/filenames.py

```diff
@@ -126,8 +126,39 @@ def determine_file_name(file_path: pathlib.Path, defaults: Dict[str, Any], overr
     :param Dict[str, Any] overrides: Name properties that should override anything detected in the file
     :raises NotImplementedError: For files that we should support but currently don't (bia, iox, obx, sum, tro)
     :return str: Proposed IGS long filename
+    """     
+    name_properties = determine_file_props(file_path, defaults, overrides)
+    return generate_IGS_long_filename(**name_properties)
+
+
+def determine_file_props(file_path: pathlib.Path, defaults: Dict[str, Any], overrides: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
`defaults` and `overrides` can and probably should default to empty dictionaries (this is true for the parent function as well). To avoid mutable-default issues it probably then needs a `defaults = None` `if defaults is None: defaults = {}` idiom.
In hindsight the type can probably just be a `Optional[Mapping[str, Any]]` and I can't remember whether that `Any` can be tightened up further.

Reply:
Updated